### DISCUSSION TITLE
Tips on using with a GraphQL endpoint?

### DISCUSSION THREAD

**Discussion by andybywire** [PavelLaptev/JSON-to-Figma#38] [Q&A]
Thank you for this plugin!—this is a great tool for content-first/content-driven design.
I have a question about using data from and API (“data from clipboard link”). I’m pulling in data from a headless CMS (Sanity), and whether I use [Sanity’s standard HTTP API](https://www.sanity.io/docs/http-api) endpoint, or the [GraphQL API](https://www.sanity.io/docs/graphql#ee2faf8b5ae3), the data always comes back as nested in a “data” or “result” object, which puts the properly formatted array of objects JSON-to-Figma needs one (or two) object(s) too far away:

```
{
  "data": {
    "allMethod": [
      {
        "title": "Card Sorting",
        "metaDescription": "Card sorting is a method used to help design or evaluate the information architecture of a site."
      }, ...
```

Here’s a simple query from each of these endpoints to show you what I mean:

`https://4g5tw1k0.api.sanity.io/v2021-03-25/data/query/production?query=*[_type==$type]{title,metaDescription}&$type="method”`

`https://4g5tw1k0.api.sanity.io/v1/graphql/production/default?query={allMethod{title,metaDescription}}
`

I have _not_ ruled out user error on my part in getting this to work, so here are two questions:
1. Am I being obtuse and missing something obvious that would get this to work with live content-as-data with the plug as-is? 
2. If this is a limitation of the plugin (as opposed to an oversight of mine), is this functionality that could be supported in a future version of the plugin? The ability to pull content data directly from a headless CMS API or GraphQL endpoint strikes me as a super useful use case for this tool, and it seems like a top level object is pretty common for these kinds of endpoints.

I’d be happy to add a “show and tell” example or make some suggestions for the docs if there’s a solution for this that I just wasn’t able to get the first time around. Thanks again for all the great stuff this tool already does!

**Comment by gejoreni** [comment]
Did you figure anything out here? I ran into the same issue and would love some guidance - digging a little it looks like the plugin is looking for an Array but gets an object? I haven't been able to get any other plugins to work either.

**Reply by andybywire** [reply]
I wasn't able to work through getting _this_ plugin to work, but I did find [Data Sync](https://www.figma.com/community/plugin/784880696244762700/Data-Sync-(previously-API%2FJSON-Sync)), which I think offers the flexibility I was looking for. I was past the point in the project where I could really run it through the paces (I ended up just copying content in), but after some light experimentation it looked promising.
Hope that helps!
